src/core/qgsvectorlayer.cpp

```diff
@@ -4445,3 +4452,27 @@ bool QgsVectorLayer::readExtentFromXml() const
 {
   return mReadExtentFromXml;
 }
+
+void QgsVectorLayer::setRefreshOnNotifyEnabled( bool enabled )
+{
+  if ( !dataProvider() )
+    return;
+
+  if ( enabled and not isRefreshOnNotifyEnabled() )
```

Comment:
Use c++ operators && ! (don't use the `and` and `not` keywords) (also in the else condition)

Reply:
sorry, ther is too much python in my life.